packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -716,6 +751,13 @@ class AndroidDevice extends Device {
       Uri? vmServiceUri;
       if (debuggingOptions.buildInfo.isDebug || debuggingOptions.buildInfo.isProfile) {
         vmServiceUri = await vmServiceDiscovery?.uri;
+        // For apps that use the VM service protocol, we *do* have a signal that the app
+        // appears to have started, and can wait until the VM service is discovered, and
+        // then check "startupLogs" to see if an error was detected so far.
+        if (checkErrorFromStartupLogs() case final String error) {
```

Comment:
We could concievably get a successfull connection but still see an error log right?

Could we ignore this error if we have a vm service URI?

Reply:
The reverse is also true though, we get a VM service URI when we otherwise don't have a successful connection :-/ (fatal Impeller)

Reply:
Ah, so we connect but then get the connection is closed error. Yeah, I dunno. Maybe you could do something like set an error bit and if an http exception is encountered then assume the app crashed. But if that is happening way later its not going to be possible to track.

I don't think there is going to be a foolproof method here. Unless we want to add some magic string to these known errors that would be less likely to colide than "Error"